game/gbtd_gbr0.ksy

```diff
@@ -0,0 +1,230 @@
+            type: u2
+          - id: split_height
+            type: u2
+          - id: split_order
+            type: u1 # Note: Official specification incorrectly says "long"
+          - id: color_set
+            type: u1
+          - id: bookmarks
+            type: u2
+            repeat: expr
+            repeat-expr: 3
+          - id: auto_update
+            type: u1
+        
+      tile_export:
+        seq:
+          - id: tile_id
+            type: u2
+          - id: file_name
+            type: strz
+            size: 128
+          - id: file_type
+            type: u1
+          - id: section_name
+            type: strz
+            size: 20
+          - id: label_name
+            type: strz
+            size: 20
+          - id: bank
+            type: u1
+          - id: tile_array
+            type: u1
+          - id: format
+            type: u1
+          - id: counter
+            type: u1
+          - id: export_from
+            type: u2
+          - id: export_to
+            type: u2
+          - id: compression
+            type: u1
+          - id: include_colors
+            type: u1
+          - id: sgb_palettes
+            type: u1
+          - id: gbc_palettes
+            type: u1
+          - id: make_meta_tiles
+            type: u1
+          - id: meta_offset
+            type: u4
+          - id: meta_counter
+            type: u1
+          - id: split
+            type: u1
+          - id: block_size
+            type: u4
+          - id: sel_tab
+            type: u1
+
+      tile_import:
+        seq:
+          - id: tile_id
+            type: u2
+          - id: file_name
+            type: strz
+            size: 128
+          - id: file_type
+            type: u1
+          - id: from_tile
+            type: u2
+          - id: to_tile
+            type: u2
+          - id: tile_count
+            type: u2
+          - id: color_conversion
+            type: u1
+          - id: first_byte
+            type: u4
+          - id: binary_file_type
+            type: u1
+
+      palette:
+        seq:
+          - id: colors
+            type: u4
+            repeat: expr
+            repeat-expr: 4
+
+      palettes:
+        seq:
+          - id: id
+            type: u2
+          - id: count
+            type: u2
+          - id: colors
+            type: palette
+            repeat: expr
+            repeat-expr: count
+          - id: sgb_count
+            type: u2
+          - id: sgb_colors
+            type: palette
+            repeat: expr
+            repeat-expr: sgb_count
+
+      tile_pal:
+        seq:
+          - id: id
+            type: u2
+          - id: count
+            type: u2
+          - id: color_set
+            type: u4
+            repeat: expr
+            repeat-expr: count
+          - id: sgb_count
+            type: u2
+          - id: sgb_color_set
+            type: u4
+            repeat: expr
+            repeat-expr: sgb_count
+        
+      deleted:
```

Comment:
It is not strictly necessary to have each value in `switch-on` matched by values in `cases`.

Reply:
Sorry, I don't fully understand. Do you mean that I can just remove "type::deleted: deleted" from the switch and it will just not be parsed?

Reply:
>Do you mean that I can just remove "type::deleted: deleted" from the switch and it will just not be parsed?

Not quite, it'd be left as byte buffer. But I guess it usually should have zero size when this type is used.

Reply:
Ah yeah, a byte buffer is fine. That is what I meant. That's what happens now as well, right? It doesn't _have_ to be 0 size, the format is designed to be able to handle items just marked as deleted but still there, almost like when you have orphan zip objects in a zip file. I don't really think it's a relevant feature for modern computers but that was the idea I think.

Either way, it sounds like a good suggestion. I will remove that line and the corresponding dummy type.

Reply:
>It doesn't have to be 0 size, the format is designed to be able to handle items just marked as deleted but still there, almost like when you have orphan zip objects in a zip file. I don't really think it's a relevant feature for modern computers but that was the idea I think.

It may make sense to keep it as it is then (because otherwise they will waste memory) or make it switcheable (an instance or a param in the root to swith it on/off and a type switching on it (can be accessed using `_root`).